### PR TITLE
Allows user to select gcc instead of clang for linking

### PR DESCRIPTION
**Pull request by meow464**

### CODE REVIEW

**Comment by certik** [comment]
Perfect! Can you please resolve the conflicts in `src/bin/lfortran.cpp`?

**Comment by meow464** [comment]
I forgot to click "Update Branch".

File: src/bin/lfortran.cpp [hunk 1]
```diff
@@ -1166,15 +1166,25 @@ int link_executable(const std::vector<std::string> &infiles,
...
             std::string options;
             std::string runtime_lib = "lfortran_runtime";
-            if (compiler_options.target != "") {
-                options = " -target " + compiler_options.target;
-            }
+
+            if (link_with_gcc)
+              CC = "gcc";
+            else
+              CC = "clang";
```
Comment (certik) [inline src/bin/lfortran.cpp:1177]: Style suggestion: Use {} and use 4 spaces:
```suggestion
            if (link_with_gcc) {
                CC = "gcc";
            } else {
                CC = "clang";
            }  
```

File: src/bin/lfortran.cpp [hunk 2]
```diff
@@ -1166,15 +1166,25 @@ int link_executable(const std::vector<std::string> &infiles,
...
+	    
+            char *env_CC = std::getenv("LFORTRAN_CC");
+            if (env_CC) CC = env_CC;
+
+            if (compiler_options.target != "" && link_with_gcc)
+              options = " --target " + compiler_options.target;
```
Comment (certik) [inline src/bin/lfortran.cpp:1183]:
```suggestion
            if (compiler_options.target != "" && link_with_gcc) {
                options = " --target " + compiler_options.target;
            }
```

File: src/bin/lfortran.cpp [hunk 3]
```diff
@@ -1166,15 +1166,25 @@ int link_executable(const std::vector<std::string> &infiles,
...
+
+            if (compiler_options.target != "" && !link_with_gcc)
+              options = " -target " + compiler_options.target;
```
Comment (certik) [inline src/bin/lfortran.cpp:1186]:
```suggestion
            if (compiler_options.target != "" && !link_with_gcc) {
                options = " -target " + compiler_options.target;
            }
```

**Comment by certik** [comment]
I think this looks great, thanks!

Let me know if things work for you regarding cross-compiling.

**Comment by meow464** [comment]
So far it seems to work. I will make more pull requests in the following weeks.

**Comment by certik** [comment]
Perfect! Can you send a PR to document how to use LFortran to cross-compile? How do you install LLVM for cross-compilation?

**Comment by meow464** [comment]
Of course! I wasn't planing on doing this only for myself, what got me here was trying to improve scipy and numpy support on kivy and beeware.

**Comment by certik** [comment]
Awesome. We are working hard on getting SciPy working well with LFortran.

**Pull request merged**